src/wallet/api/wallet2_api.h

```diff
@@ -1144,6 +1154,519 @@ struct Wallet
+// TODO : wallet2::signed_tx_set & cryptonote::address_parse_info
+    /**
+    * brief: coldSignTx -
+    * param: ptx -
+    * param: exported_txs -
+    * param: dsts_info -
+    * param: tx_device_aux -
+    */
+//    virtual void coldSignTx(const PendingTransaction &ptx, signed_tx_set &exported_txs, std::vector<cryptonote::address_parse_info> &dsts_info, std::vector<std::string> & tx_device_aux) const = 0;
+    // TODO : this already exists, remove after checking
+    /**
+    * brief: useForkRules - check if the wallet uses the rules for hard fork version at current block height - early_blocks
+    * param: version - hard fork version
+    * param: early_blocks -
+    * return: true if fork rules are used
+    * note: sets status error on fail
+    */
+//    virtual bool useForkRules(std::uint8_t version, std::int64_t early_blocks) const = 0;
+// TODO : wallet2::transfer_details
+    /**
+    * brief: getTransferDetails -
+    * param: idx - index in transfer storage
+    * return: transfer details for given index
+    * note: sets status error on fail
+    */
+//    virtual const wallet2::transfer_details &getTransferDetails(std::size_t idx) const = 0;
+    /**
+    * brief: discardUnmixableOutputs - freeze all unmixable outputs
+    * note: sets status error on fail
+    */
+    virtual void discardUnmixableOutputs() = 0;
+    /**
+    * brief: setTxKey - set the transaction key (r) for a given <txid> in case the tx was made by some other device or 3rd party wallet
+    * param: txid -
+    * param: tx_key - secret transaction key r
+    * param: additional_tx_keys -
+    * param: single_destination_subaddress -
+    * note: sets status error on fail
+    */
+    // QUESTION : There were some cases already where I wished to have boost::optional in here. I have no explicit example, but in general e.g. for functions that return a bool, but have to return early on error, it would be cleaner to return boost::none instead of true/false, because the API user could forget to check the error status and may use an incorrect value. Similar for uint*_t, where we can't use e.g. -1 on error.
+    //            If we add access to boost::optional revisit all new functions to see which would benefit from it.
+    virtual void setTxKey(const std::string &txid, const std::string &tx_key, const std::vector<std::string> &additional_tx_keys, const boost::optional<std::string> &single_destination_subaddress) = 0;
+    /**
+    * brief: getDaemonAddress -
+    * return: daemon address
+    */
+    virtual std::string getDaemonAddress() const = 0;
+    /**
+    * brief: getDaemonAdjustedTime -
+    * return: daemon adjusted time
+    * note: sets status error on fail
+    */
+    virtual std::uint64_t getDaemonAdjustedTime() const = 0;
+    /**
+    * brief: setCacheDescription - set wallet cache attribute with key ATTRIBUTE_DESCRIPTION
+    * param: description -
+    */
+    virtual void setCacheDescription(const std::string &description) = 0;
```

Comment:
IMHO using the term *cache* here is overly technical: Seems to me you simply set the *wallet description* with this.

Reply:
Actually, since we want to keep things minimal and we already have `setCacheAttribute` ([here](https://github.com/monero-project/monero/blob/a1dc85c5373a30f14aaf7dcfdd95f5a7375d3623/src/wallet/api/wallet2_api.h#L1013)) and `wallet2::set_description` (the function that was intended to get replaced by `setCacheDescription`) just calls `wallet2::set_attribute(ATTRIBUTE_DESCRIPTION, description)` I think we can remove this and the get function below.